### DISCUSSION TITLE
Sprint 13

### DISCUSSION THREAD

**Discussion by Justus-at-Tazama** [tazama-lf/General-Issues#305] [General]
## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
None

### @cshezi - Cebolenkosi Shezi

**Yesterday:**

**Today:**

### @vorsterk - Kyle Vorster

**Yesterday:**

**Today:**

### @Lenbkr - Len Bekker

**Yesterday:**

**Today:**

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:**

**Today:**

### @rtkay123 - Rodney Kanjala

**Yesterday:**

**Today:**

### @johanfol - Johan Foley

**Yesterday:**

**Today:**

**Comment by Justus-at-Tazama** [comment]
## 2023/10/23

## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
None

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:**

**Today:** Final day of product training

### @vorsterk - Kyle Vorster

**Yesterday:** Enabled log-stash and APM and redeployed to identify any bottle-necks. Core DNS were not performing optimally - experimenting with some options. Got up to 1750 evaluations per second.

**Today:** Push hard. Like a Springbok scrum. Get a higher score than 1 better than the opposition. Demo prep.

### @Lenbkr - Len Bekker

**Yesterday:** Looked into configurating cloud agents for Kubernetes. Could not initially get it to start up. Even though URLs and credentials were correct, something else was failing. Kubernetes/Azure URL is apparently not resolving properly. Once this is resolved, you should in theory be able to automatically spin up the Kubernetes builder nodes.

**Today:** Carry on trying to resolve the Kubernetes/Azure URL. May also be permissions. Demo prep and video. Need to catch up on some documentation.

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Sybrin SCM update; reviewed with Rodney and Shezi - fixed a couple of issues and hardened the app. Robustness needed some work for validations via a wizard. Miscellaneous fixes out of the demo on Wednesday. Investigated the "timaframs" issue - the Bun switch-back left a gap and every rule was deployed as rule 901. Now fixed and every rule is now deployed.

**Today:** Ran a test and there is some weirdness related to the network map - seems to be an issue in the network map build. Justus and JP to review this morning and wrap up. Demo video and prep.

### @rtkay123 - Rodney Kanjala

**Yesterday:**

**Today:** Final day of product training

### @johanfol - Johan Foley

**Yesterday:** Performance testing with Kyle.

**Today:** Assist Kyle with hardware scaling. Investigating a better way to implement logging.

## Parking lot

**Comment by Justus-at-Tazama** [comment]
## 2023/10/24

## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@rtkay123 - Rodney Kanjala

## Apologies
@vorsterk - Kyle Vorster

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Met with JP and Rodney to review the SCM. Also review: https://frmscoe.atlassian.net/wiki/spaces/FRMS/pages/76906497/Configuration+management.

**Today:** Investigate gaps in the SCM and then start looking at reingestion of the Bank X data.

### @vorsterk - Kyle Vorster

**Yesterday:** Performance benchmarking

**Today:** Getting better.

### @Lenbkr - Len Bekker

**Yesterday:** Admin and ticket creation catch-up. Demo preparation.

**Today:** Wrap up Jenkins Cloud agents on SB1.

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Fixed and redeployed all the rules on SB1. Ran into a bug in the TADProc where the TADProc was using the transaction configuration to check channel composition instead of the network map. Bug addressed addressed and will be going live today.

**Today:** Wrap up channel resolution and redeploy to also include Rodney's description and reason code fix. Then testing, then deployment to SB2 as well (Shezi).

### @rtkay123 - Rodney Kanjala

**Yesterday:** Implemented the change to reduce the text in the rule result. The additional information in the rule result is stripped out by the rule executor. Sat with JP to review the SCM.

**Today:** Complete Sybrin product certification.

### @johanfol - Johan Foley

**Yesterday:** Performance benchmarking

**Today:** Design and architecture, planning.

## Parking lot

Design discussions:
 - List management
 - Dynamic routing via the network map
 - Centralized logging
 - Dynamic scaling

**Comment by Justus-at-Tazama** [comment]
## 2023/10/25

## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@rtkay123 - Rodney Kanjala

## Apologies
@vorsterk - Kyle Vorster

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Wrapping up training assessments. Deploy the deployment manager to access the Sybrin Configuration Manager. Some issues, resolved.

**Today:** Waiting for testing on Main to be complete to start deploying on SB2. Working on some outstanding features and issues on SCM.

### @vorsterk - Kyle Vorster

**Yesterday:** Sick leave

**Today:** Sick leave

### @Lenbkr - Len Bekker

**Yesterday:** Jenkins cloud agents that leverage Kubenetes. Got Jenkins to connect using admin privileges. Dynamically spins up when a build is schedules and shuts down when complete. The build instructions themselves fail though.

**Today:** Review the build template to see what is missing/broken. Review Jenkins plug-ins. Troubleshoot and experiment. Foley suggests checking in with Jason/DevOps team.

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** So much fun troubleshooting SB1. Fixed test, ran into a bug, fixed that too. Unnecessary loops in the channel processing. Deployment manager troubleshooting and fix.

**Today:** Found a bug, fixed a bug, missing typologies. Debugging and troubleshooting continues.

### @rtkay123 - Rodney Kanjala

**Yesterday:** Completed training assessment. Deployed new rules on SB4.

**Today:** Update documentation, crossing t's, dotting i's.

### @johanfol - Johan Foley

**Yesterday:** Design authority prep and project planning.

**Today:** Research  and planning on centralised logging and dynamic scaling.

## Parking lot

**Comment by Justus-at-Tazama** [comment]
## 2023/10/09

## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
None

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Trying to get the SCM onto local. Supported by Foley and Rodney. Tried to deploy with the deployment manager packaged with the platform. Gets stuck on 90% all the time. Tried the local Deployment Manager which seemed to work better. Tried to restore the snapshot into the local environment, the changes are applied on the side of the Deployment Manager and... chaos.

**Today:** Need a new ticket.

### @vorsterk - Kyle Vorster

**Yesterday:** Getting better

**Today:** Getting better

### @Lenbkr - Len Bekker

**Yesterday:** Jenkins Cloud Agent - Got it to work in a sense where it can fully spin up, but have not been able to get it to build the full platform from a custom pipeline. Called out to the Sybrin DevOps team for support.

**Today:** Carry on and hope for a Jenkins Cloud silver lining. Lots of experimentation in the blind - hopefully the Sybrin team can assist. 

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Fixed end-to-end. One outstanding issue around the propagation of the new workflow object out of the typology processor.

**Today:** Propagation of the workflow object, but mostly trying to get better.

### @rtkay123 - Rodney Kanjala

**Yesterday:** Reviewed and updated documentation. Clarified functionality around descriptions and reason code utilisation. Supported Shezi on the deployment manager. Updated the deployment pipeline scripts. Reviewed the deployment process for gaps - nothing evident at the moment - move the ticket to done. Also struggling to deploy a snapshot of the Sybrin Configuration Manager.

**Today:** Look for a way to improve the private rules into the Redactio Docker.

### @johanfol - Johan Foley

**Yesterday:**

**Today:**

## Parking lot

**Comment by Justus-at-Tazama** [comment]
## 2023/10/27

## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
None

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Still attempting to deploy the SCM on local. Able to successfully deploy on local. Still missing the "tree view" - calling on JP to assist. Exploring alternative ways of visualization of the configuration in the SCM.

**Today:** Import the plug-in and working on improving and fixing the SCM.

### @vorsterk - Kyle Vorster

**Yesterday:** Been working on the performance benchmarking. No matter how many VMs we tried, we couldn't get any better. Jason suggested benchmarking Redis in isolation to make sure there's no bottleneck there. Current max is about 1800 evaluations per second.

**Today:** Resume performance benchmarking.

### @Lenbkr - Len Bekker

**Yesterday:** Worked through the Jenkins Cloud Agent with the internal Sybrin DevOps team. Some learnings, but still not fully resolved. But wait! Eventually able to build a full job from a Jenkins pod start-to-finish. The build took a bit longer than Docker. Still need to tweak the pod resources. Still need to attempt multi-queue builds. Build is less than 10 minutes.

**Today:** Test multi-build and resource tweaking; replicate to dev once satisfied.

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Investigated the case of the missing workflow. The Typology Processor is now sending the object, but it is not arriving in the TADProc. Please help. May be a protobuf issue.

**Today:** Find the missing workflow object.

### @rtkay123 - Rodney Kanjala

**Yesterday:**

**Today:** HAPPY BIRTHDAY!

### @johanfol - Johan Foley

**Yesterday:** 

**Today:** Planning and stuff. Meet Justus at 14:00.

## Parking lot

As soon as JP is done with the workflow object propagation, we'll cut release 1.2.0.
Justus to share testing approach.
No new features into main until the release is built.

**Comment by vorsterk** [comment]
## 2023/10/30

## Attendees
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@johanfol - Johan Foley

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Worked on integrating tree view for SCM to provide another view for hierarchy view of network configuration and making sure it function similar to graph view already implemented.

**Today:** Work on adding JP plugin for graph view and start error handling that present message to the user.

### @vorsterk - Kyle Vorster

**Yesterday:** Resumed performance benchmarking for 300 FTPS goal , not seeing results , doing research on my to optimize redis due to the more redis instances seems to slow down the performance runs

**Today:** check on the release builds and do tests to cut a release. (ie: waiting for Justus for approval) 

### @Lenbkr - Len Bekker

**Yesterday:** Took some trickery but I was able to stabilize the jenkins builders from not failing so often, done by tweaking concurrency and resource quotas as well as buildah build arguments. Tested up to 3 jenkins builders at the same time with great success. Apparently only owners can assign azure containers to resources so not even Kyle can do it for the dev environment. Will ask Donald today.

**Today:** Update dev environment with jenkins pod builders

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Finished Testing with justus on the Typology Processor and TADP, Found the missing link causing the workflow to disappear

**Today:** Picked up new logging ticket.

### @rtkay123 - Rodney Kanjala

**Yesterday:** HAPPY BIRTHDAY!

**Today:** Going to start upskilling in angular

### @johanfol - Johan Foley

**Yesterday:** Mojaloop Convening 

**Today:** Mojaloop Convening

## Parking lot

**Comment by vorsterk** [comment]
## 2023/10/31

## Attendees
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@johanfol - Johan Foley

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Worked on getting the graph view plugin to work on SCM but still no luck seems like nexus auth problems had support of JP and Rodney, and then tried to help Rodney setup his local SCM environment but they were obstacles, continued trying to error handle more from SCM.

**Today:** Continue working on SCM and hit the getting started of angular from the course that was recommended by Foley, The plan is spitting the days in half , first half working on SCM and other half up skilling myself.

### @vorsterk - Kyle Vorster

**Yesterday:** Looking at the release build , created another repo to run a test to make sure everything is still working from 3 months ago if and changes affected the release builds . Github token substitutin was giving me grief all of a sudden.

**Today:** Looking at Github token substitution problem , If Justus is happy cut the release and help Len with testing the piepline.

### @Lenbkr - Len Bekker

**Yesterday:** Configured Cloud on Dev Jenkins, Started with Processors to see if I can get it to work. You'd think that because everything was working on the sandbox it should be as setting mostly the same shit on dev env. Nope, here new issues at every step. Anyway I got through the processors except for the Rust one which I'm skipping investigating for now. Shout out to Donald for attach container to dev env as well. Made my life little easier.

**Today:** Still need to update each and every rule on Dev and then do a big pipeline test with Kyle.

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Worked on the Logging Ticket , Currently testing with a single processor, with the logging changes in the library mainly. Getting filebeats up and running.

**Today:** Continue with Ticket.

### @rtkay123 - Rodney Kanjala

**Yesterday:** Support from JP for Graph Visualiser plugin. Tried to get local testing for plugin on Shezi's environment. Tried to use snapshot from SB3 on local env, almost finished but eskom cables "disappeared". A bit of angular skill up

**Today:** Find some electricity, finish restoring checkpoint and continue angular skill up.

### @johanfol - Johan Foley

**Yesterday:** Mojaloop Convening 

**Today:** Mojaloop Convening

## Parking lot

**Comment by JeanPierreNell** [comment]
## 2023/11/01

## Attendees
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@johanfol - Johan Foley

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Townhall meeting. Used some of the day for upskilling in Angular, continued with the CSM, Justus popped a message about redeploying and ingesting the data. Did try and redeploy everything, but e2e is not up and running yet. Tried to trim it down to a single typology and a single rule. Typology IDs are messed up. 

**Today:** Will contact JP or Justus for TP ids

### @vorsterk - Kyle Vorster

**Yesterday:** Trying to build release, Justus did approve current code. Cloned TP repo, ran a few tests, kept running into issues with the milestone workflow files, on the release workflows as well. .npmrc file was missing. Checked Foley's workflow changes to a node.js workflow file. Had an issue with the tests, rodney to the rescue with parameter. Started creating the PRs for all the rules and rule-executor. 

**Today:** Needs approval for rule PRs , cutting release. Carry on with Performance benchmarking, the more redis we have, the slower our throughput. Makes no sense. Wil talk to Jason.

### @Lenbkr - Len Bekker

**Yesterday:** Implemented jenkins builders for the rules themselves on dev. Previously only on processors. Currently in a working order. Speculation has arised when doing it this way, they won't have any cache, every builder will have to pull from docker hub. Azure might have a solution for this, but it's not great, since we might be rate limiting our docker URLs. Second optimization is Jenkins cloud builders is maxing out. Need a discussion with kyle surrounding the location of these builders.

**Today:** Chat with Kyle. Need to create new ticket for the docking caching issue. (Don't wanna DDOS Dockerhub)

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Worked on implementing logging change on library, built a local version and installed in TP - currently configuring filebeats for elastic on local env.

**Today:** Get Filebeats working, e2e testing with logging

### @rtkay123 - Rodney Kanjala

**Yesterday:** Finished setting up local platform environment. Tried to deploy graph-visualizer plugin, got stuck. Got a message from JP on monday about updating SB4 configuration - e2e broke on SB4. Was on a call with Shezi this morning to figure out what was going on - just needed to redeploy TP

**Today:** Must watch Angular videos - will run and avoid it, break local environment - bothering Shezi for platform stuff. Take a crack at default values on SCM. 

### @johanfol - Johan Foley

**Yesterday:** Mojaloop Convening
 
**Today:** Mojaloop Convening

## Parking lot

**Reply by rtkay123** [reply]
## 2023/11/02

## Attendees
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@johanfol - Johan Foley

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Kickstarted batch processing and ingestion. Took 7 hours, one transaction missed. Justus is sus with his testing and may have caused interference. Continued with SCM cfg. Currently checking if integrated view works across system. Some issues, but will try continue over with local env  

**Today:** Local impls over to remote. (Angular upskill pending)

### @vorsterk - Kyle Vorster

**Yesterday:** Working on cutting a release. Spoke to Foley -> make changes to remove version suffixes from some Dockerfiles (`-rel-1-0-0`), they shall be populated in jenkins pipelines. Updating versions in `package.json` with `typology-processor`. Some tags aren't being pulled for releases, investigated typology-processor.

**Today:** Investigate release version for tags so they can be pulled on Jenkins. Have some stuff in the works, Len to review. 

### @Lenbkr - Len Bekker

**Yesterday:** Adding Azure cache (caching layer for docker containers we're pulling through, just node for now, F for distroless). Updated all rules and processors, tested on dev. Caching works but there are some edge cases that need intervening.

**Today:** Investigate how to get distroless working with cache. Connect with Kyle to test pipelines 

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Got Filebeats working. Connected to Elastic. Filebeats doesn't support subscribing to NATS subjects :(. Some investigation, but seems NATS logs can be read, just not the streams. 

**Today:** Investigating Filebeats interop with NATS streams

### @rtkay123 - Rodney Kanjala

**Yesterday:** Updating Rust typology processor. tested with local deployed redis cluster. Managed to reproduce issue on Dev. Managed to find resolution. Fixed some migration issues with Deployment Manager. Local environment cannot generate reports. Investigating why

**Today:** Configure default values for platform. Skill up


### @johanfol - Johan Foley

**Yesterday:** Mojaloop Convening
 
**Today:** Mojaloop Convening

## Parking lot

**Comment by cshezi** [comment]
## 2023/11/03

## Attendees
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@johanfol - Johan Foley

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Angular Up-skilling and Tried to install graph visualizer but suspect something auth with nexus because of `npm i ` not installing the Sybrin-related libraries

**Today:** Continue with Angular upskilling and try what guys suggest with `npm i` of the graph-visualizer plugin

### @vorsterk - Kyle Vorster 

**Yesterday:** Completed all the releases (Yey) still working on Jenkins pipelines worked with Len, replicated Jenkins builds, tested TMS, one variable can be changed to specify if tag or branch to target Cool stuff

**Today:** Continuing with the Jenkins job.

### @Lenbkr - Len Bekker

**Yesterday:** Distroless tested and it works Yey, need to align the rest of the jobs, approve some of Kyles's PRs

**Today:** Finishing image caching things and looking for new tickets

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Checked feasibility with Foley, Foley checked with Node people, Len checked with JP another option called vector it was promising and Foley removed vector as an option, and suggested batching the logs and sending them to ELK

**Today:** New Logging tickets

### @rtkay123 - Rodney Kanjala

**Yesterday:** A Bunch of Angular Practice, tried to deploy the graph visualizer on the local instance of SCM but still no luck, something with WSL and Windows bridge connection

**Today:** Angular upskilling and try to connect WSL with windows

### @johanfol - Johan Foley

Yesterday: Mojaloop Convening

Today: Mojaloop Convening

## Parking lot

**Comment by vorsterk** [comment]
## 2023/11/06

## Attendees
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@johanfol - Johan Foley

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Try target packages for the lib that Auth failed (depth keep) . Still getting an error, tried moving more packages to see if that sorts out the Auth issue. Up skilling training (10 videos) . New courses to work on

**Today:** Upskilling courses again today .

### @vorsterk - Kyle Vorster 

**Yesterday:** Worked on building the new release jenkins pipelines plus changing all the rules and processors to match.

**Today:** Will be speaking to Justus to see if he wants me to test the release configuration or if he will when he gets back , If not I will be looking at the redis bench marking setup on DEV to run a few tests to see if replication actually effects performance . Speak to Justus about tasks for Rodney and Len . Add caching layer on jenkins from Lens stuff , and clean up deployment jobs.

### @Lenbkr - Len Bekker

**Yesterday:** Updated the Jenkins jobs with the caching . Done queue builder jobs , testing and working. 

**Today:** Looking for a ticket , base pictures ticket ...maybe?

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Building an event side car for logging and logging service , changes to protofiles 

**Today:** Carry on with the logging service and event sidecar..

### @rtkay123 - Rodney Kanjala

**Yesterday:** Angular upskilling (courses) and try to connect WSL with windows , also worked on graph visualizer . 

**Today:** Angular upskilling and maybe new ticket if there is one to be picked up.  Take a look at PR

### @johanfol - Johan Foley

Yesterday: Mojaloop Convening

Today: Mojaloop Convening

## Parking lot

JP: Harding for Sybrin SCM - no ticket but work could be done for this

**Comment by JeanPierreNell** [comment]
## 2023/11/07

## Attendees
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@johanfol - Johan Foley

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Continued with Angular Upskilling. Completed Getting Started Course. Asked Jason for the subscription for training, but it's in use until next month. Asked Jason about the authentication around nexus, seems like there is none. Jason suggested taking a look at the onboarding document. Need to find document. 

**Today:** New Tickets. Maybe upskilling. Practice Angular. Nexus?

### @vorsterk - Kyle Vorster

**Yesterday:** Worked on generating payload for redis through benchmarking. Bit of a nightmare. Reading into pipeline vs. scripting. 

**Today:** Running the tests. Need to reinstall the helm charts for redis, cant scale up on the fly. Want to run tests on alternative TPs tomorrow. TP is our most intensive processor, should effect results.

### @Lenbkr - Len Bekker

**Yesterday:** Took on base fixture ticket. If it doesn't live in it's own repo, it will live with the library. Maybe get networkhelper function, static messages. 

**Today:** Continuing with that ticket today. 

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Continued on with logging ticket, had some grpc communication issues and npm building issues. Building is now resolved. Built Proto files.

**Today:** Continue with logging.

### @rtkay123 - Rodney Kanjala

**Yesterday:** Recap of angular course. Memory 404, reimplemented stuff, revisit the course when stuff was foggy. rxjs ones are locked behind a paywall. Got youtube videos for rxjs. 

**Today:** Finishing up the txjs , finding some angular exercises somewhere. "Kyle: JUST DO IT".

### @johanfol - Johan Foley

**Yesterday:** Convening 

**Today:** Convening

## Parking lot

Kyle: Sent Justus a message yesterday, but no results as of yet.

**Comment by rtkay123** [comment]
## 2023/11/08

## Attendees
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@johanfol - Johan Foley

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Continued with `RxJs` practice/skill-up. Simultaneously searching for activities that would put skills to practice. No findings, going to freestyle it. Took a course for creating/managing plugins - completed it. 

**Today:** Continuing with `RxJs` practice and moving to `NgRx`. More practical applications

### @vorsterk - Kyle Vorster

**Yesterday:** Working on Redis benchmarking, created a [doc](https://frmscoe.atlassian.net/wiki/spaces/FRMS/pages/86507521/Redis+Benchmarking+Test+Plan) for it on Confluence. Ran ~8 benchmarks on 3vms. Updated pipelines 

**Today:** Generating report (Excel) for review. Updating the pipelines to run Len's Jenkins build. Performance benchmarking

### @Lenbkr - Len Bekker

**Yesterday:** Worked on base fixtures, functionality implemented on local branch, had a meeting to gather usecases, had a frontend runthrough to show where solution fits in

**Today:**
Meeting Tinus to discuss data model and other back-end stuff. Implementing use cases and app builder w/JP based on their work.

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Continued with logging implementation. Put on pause as other fraud risk thing got prioritised. 

**Today:** Meeting with Tinus, going to start building, probably

### @rtkay123 - Rodney Kanjala

**Yesterday:** Finishing `RxJs` course. Looked for exercises to put course learnings into practice. Got a recommended course from @cshezi for building plugins.

**Today:** `NgRx` course and taking 'plugin builder' course. Exploring code base to see if we can understand what's going on a little better.

## Parking lot
(there was talk of pizza in the morning)

**Comment by cshezi** [comment]
## **2023/10/09**
Attendees

@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

Apologies
@johanfol - Johan Foley

Minutes:
@cshezi - **Cebolenkosi Shezi**
Yesterday:  Continued with angular up-skilling mainly rxjs and started looking at ngrx, found a nice code sandbox for practicing and just revising all the learned concepts of angular.

Today: Continue with ngrx and maybe start to look at the code from Sybrin platform repo if given access that was requested by Rodney yesterday.

@vorsterk - **Kyle Vorster**
Yesterday: Redis cluster benchmarking, Jason suggested running it with pipelining, the activity has been paused, did Benchmark rust tp, no more dupes got an error and Got Rodney to look at it, Rust tp might give us huge performance gain, adding a repository for JP.

Today:  Working with Rodney to run that rust TP and working on Jupiter stuff to get graphs and Redis benchmarking.

@Lenbkr - **Len Bekker**
Yesterday: Got on a call with JP and Tinus to walk through the solution, and with JP did planning based on the cases Roland provided.

Today: Dummy data generation for creating demo ware for faking the payload, working with JP.

@JeanPierreNell - **Jean-Pierre Nell**
Yesterday: Set with Tinus on a meeting, Rodney took the logging tickets, pushed the protofile to the repository, completed the data model, and built screens and microflows.

Today: Working with Len on the Test data and trying to make it work.

@rtkay123 - **Rodney Kanjala**
Yesterday: DevOps access for the repo but no response yet, worked with Kyle with rust tp, walkthrough of logging service with JP, and started playing around with the concept, Problem with protofile targeting path

Today: Fixing rust tp, deploying Jupiter 

@johanfol - **Johan Foley**
Yesterday: Convening

Today: Convening

# Parking lot

**Comment by Justus-at-Tazama** [comment]
## 2023/11/10

## Attendees
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@johanfol - Johan Foley

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Updates to the Sybrin Configuration Manager. Upskilling on Angular. 

**Today:** Check in with Foley about re-running SB2. 

### @vorsterk - Kyle Vorster

**Yesterday:** Redis benchmarking. Trying to see if an increase in pods results in an increase in throughput. Worked with Rodney to get Jupyter up and running for the performance visualizations. Docker file was not building correctly off Rodney's configuration. Found an issue on the port configuration. Got it up and running and created working Helm charts for the deployment. Also created a Jenkins pipeline for the deployment. 

**Today:** Complete redis benchmarking. Test Jupyter deployment process. Performance benchmarking on Rust typology processor.

### @Lenbkr - Len Bekker

**Yesterday:** Working on pre-sales and demo stuff with JP, including test data.

**Today:** Working on pre-sales and demo stuff with JP.

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Demo preparations. Support for Rodney in handing over the work for the logging.

**Today:** Carrying on with demo preparations. Go processor on hold(ish) for now until probably Monday.

### @rtkay123 - Rodney Kanjala

**Yesterday:** Worked with Kyle to fix Jupyter Docker deployment. Prepared SQL server database - ready for new data. Started working on the logging framework - integrated NATS. Collaborating with JP as part of the handover.

**Today:** Continuing with logging.

### @johanfol - Johan Foley

**Yesterday:**

**Today:**

## Parking lot

Logging:
